src/content/core/okteto-variables.mdx

```diff
@@ -7,36 +7,259 @@ id: okteto-variables
+    terraform plan -var-file=./config/${ENV}/terraform.tfvars -out tfplan
+    terraform apply tfplan
+```
+```yaml
+# okteto.yml
+deploy:
+  commands:
+    - make deploy
+```
+
+
+
+### Sharing variables between commands with $OKTETO_ENV
+
+As mentioned in the section [Scope](#scope), variables defined in a specific command item are only available to that item by default.
+
+We understand that projects of a certain size and maturity often need to automate such steps, and often that require sharing values between steps. For such scenario we recommend using the special variable called `$OKTETO_ENV`.
+
+```yaml
+test:
+  commands:
+    # Example: some infrastructure has been provisioned
+    - terraform init -backend-config=./config/${ENV}/backend.hcl
+    - terraform plan -var-file=./config/${ENV}/terraform.tfvars -out tfplan
+    - terraform apply tfplan
+    - terraform output -json > tfout.json
+
+    # I make the following var available to other steps
+    - SERVICE_ENDPOINT=$(cat tfout.json | jq -r '.service_endpoint') >> $OKTETO_ENV
+
+    # the following step will have access to SERVICE_ENDPOINT
+    - echo "Running tests against ${SERVICE_ENDPOINT}..."
+```
+
+Another example for the same functionality, can be for example, creating the infrastructure, generating a `.env` file, and then deploy the application.
+
+```yaml
+deploy:
+  commands:
+  - name: Create AWS infrastructure
+    command: |
+      set -e
+      bucketName="${OKTETO_NAMESPACE}-okteto-tacos-shop"
+      terraform apply -input=false -var "bucket_name=$bucketName" -auto-approve
+      echo "S3_BUCKET_NAME=$resourceName" >> "$OKTETO_ENV"
+
+  - name: Create .env
+    command: |
+      set -e
+      echo "S3_BUCKET_NAME=$S3_BUCKET_NAME" > .env
+
+  - name: Deploy the app
+    command: |
+      set -e
+      make deploy
+```
+
+:::note
+
+Here are some highlights, and clarifications on the example above:
+
+1. The first step makes the variable `S3_BUCKET_NAME` available to the rest of the deployment steps. Both the step "Create .env" and "Deploy the app" will have access to `S3_BUCKET_NAME`.
+
+2. Also, it's important to remember that the variables in `$OKTETO_ENV` are not shared across `deploy`, `test`, and `destroy` sections. They are only shared within the same section.
+
+3. If you are wondering where the variable `OKTETO_NAMESPACE` comes from, check the section Okteto Variables [Managed by Okteto](#managed-by-okteto).
+
+:::
+
+
+### Configuring dynamic endpoints
+
+In this example, we show how `$OKTETO_ENV` can be used to create dynamic endpoints for external resources within your dev environment.
+
+If you are not familiar yet with the concept of Okteto external resource, image a component that is not part of the deployment of your dev environment, but that you need to interact with. For example, a database, a message queue, or an S3 bucket. If you need more details on this feature, check out [External Resources](/docs/development/external-resources).
```

Comment:
I think we can skip this paragraphn with the previous one if we link to https://www.okteto.com/docs/core/okteto-manifest/#external before